tool/modules/uds_fuzz.py

```diff
@@ -0,0 +1,473 @@
+    0x83: "ENGINE_IS_RUNNING",
+    0x84: "ENGINE_IS_NOT_RUNNING",
+    0x85: "ENGINE_RUN_TIME_TOO_LOW",
+    0x86: "TEMPERATURE_TOO_HIGH",
+    0x87: "TEMPERATURE_TOO_LOW",
+    0x88: "VEHICLE_SPEED_TOO_HIGH",
+    0x89: "VEHICLE_SPEED_TOO_LOW",
+    0x8A: "THROTTLE_PEDAL_TOO_HIGH",
+    0x8B: "THROTTLE_PEDAL_TOO_LOW",
+    0x8C: "TRANSMISSION_RANGE_NOT_IN_NEUTRAL",
+    0x8D: "TRANSMISSION_RANGE_NOT_IN_GEAR",
+    0x8F: "BRAKE_SWITCHES_NOT_CLOSED",
+    0x90: "SHIFT_LEVER_NOT_IN_PARK",
+    0x91: "TORQUE_CONVERTER_CLUTCH_LOCKED",
+    0x92: "VOLTAGE_TOO_HIGH",
+    0x93: "VOLTAGE_TOO_LOW"
+}
+
+# Number of seconds to wait between messages
+DELAY_SECSEED_RESET = 0.011
+DELAY_FUZZ_RESET = 3.901
+
+# Message data length limits
+MIN_DATA_LENGTH = 1
+MAX_DATA_LENGTH = 8
+# Max size of random seed if no seed is provided in arguments
+DEFAULT_SEED_MAX = 2 ** 16
+# Number of sub-lists to split message list into per round in 'replay' mode
+REPLAY_NUMBER_OF_SUB_LISTS = 5
+BYTE_MIN = 0x00
+BYTE_MAX = 0xFF
+
+# Duplicate testing from https://www.iditect.com/guide/python/python_howto_find_the_duplicates_in_a_list.html
+def find_duplicates(sequence):
+  first_seen = set()
+  first_seen_add = first_seen.add  
+  duplicates = set(i for i in sequence if i in first_seen or first_seen_add(i) )
+  return duplicates 
+
+def ecu_reset(arb_id_request, arb_id_response, reset_type, timeout):
+    """Sends an ECU Reset message to 'arb_id_request'. Returns the first
+        response received from 'arb_id_response' within 'timeout' seconds
+        or None otherwise.
+
+    :param arb_id_request: arbitration ID for requests
+    :param arb_id_response: arbitration ID for responses
+    :param reset_type: value corresponding to a reset type
+    :param timeout: seconds to wait for response before timeout, or None
+                    for default UDS timeout
+    :type arb_id_request: int
+    :type arb_id_response int
+    :type reset_type: int
+    :type timeout: float or None
+    :return: list of response byte values on success, None otherwise
+    :rtype [int] or None
+    """
+    # Sanity checks
+    if not BYTE_MIN <= reset_type <= BYTE_MAX:
+        raise ValueError("reset type must be within interval "
+                         "0x{0:02x}-0x{1:02x}"
+                         .format(BYTE_MIN, BYTE_MAX))
+    if isinstance(timeout, float) and timeout < 0.0:
+        raise ValueError("timeout value ({0}) cannot be negative"
+                         .format(timeout))
+
+    with IsoTp(arb_id_request=arb_id_request,
+               arb_id_response=arb_id_response) as tp:
+        # Setup filter for incoming messages
+        tp.set_filter_single_arbitration_id(arb_id_response)
+        with Iso14229_1(tp) as uds:
+            # Set timeout
+            if timeout is not None:
+                uds.P3_CLIENT = timeout
+
+            response = uds.ecu_reset(reset_type=reset_type)
+            return response
+
+
+def print_negative_response(response):
```

Comment:
This should be imported from the UDS module, rather than copy-pasted into this module.